### PR TITLE
[PyOV] Upgrade ONNX to 1.12.0

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Upgrade ONNX to 1.12.0 in Python requirements
 - Upgrade ONNX to 1.12.0 in `openvino/thirdparty/onnx`
 - Mark tests of unsupported ops/features/bugs as xfails or skips

### Tickets:
 - 90678

### CODE REVIEW

File: src/bindings/python/tests/test_onnx/utils/model_importer.py [hunk 1]
```diff
@@ -28,9 +29,10 @@
     Sequence,
 )
 
-
-# add post-processing function as part of test data
-ExtOnnxTestCase = namedtuple("TestCaseExt", OnnxTestCase._fields + ("post_processing",))
+if getattr(OnnxTestCase, '_fields', None):
+    ExtOnnxTestCase = OnnxTestCase._fields + ("post_processing",)
+else:  # for ONNX >= 1.12
+    ExtOnnxTestCase = tuple((field.name for field in dataclasses.fields(OnnxTestCase))) + ("post_processing",)
```
Comment (p-wysocki) [inline src/bindings/python/tests/test_onnx/utils/model_importer.py:35]: In ONNX < 1.12 `OnnxTestCase` is a `namedtuple`, in ONNX == 1.12 it's a `dataclass`.

File: tools/mo/requirements_onnx.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-onnx>=1.8.1,<1.12
+onnx==1.12.0
```
Comment (jiwaszki) [inline tools/mo/requirements_onnx.txt:1]: Does MO should use only one specific version? I do not know all of the internals, but supporting all from `1.8.1` to (inlcuding) `1.12.0` should be doable as MO is "semi-standalone" project. Please verify that.

File: tools/mo/requirements.txt [hunk 3]
```diff
@@ -5,7 +5,7 @@ mxnet~=1.7.0.post2; sys_platform != 'win32'
 networkx~=2.5; python_version <= "3.6"
 networkx<2.8.1; python_version > "3.6"
 protobuf>=3.18.1,<4.0.0
-onnx>=1.8.1,<1.12
+onnx==1.12.0
```
Comment (jiwaszki) [inline tools/mo/requirements.txt:8]: Does MO should use only one specific version? I do not know all of the internals, but supporting all from `1.8.1` to (inlcuding) `1.12.0` should be doable as MO is "semi-standalone" project. Please verify that.

File: tools/mo/requirements_onnx.txt (hunk 2 above)
Comment (p-wysocki) [reply tools/mo/requirements_onnx.txt]: You're right, I relaxed the requirement.

**Review by jiwaszki — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Review by akuporos — APPROVED** [review]

**Pull request merged**